packages/translated-components/src/index.js

```diff
@@ -61,16 +62,29 @@ const moneyFormat = (language) => ({
 
 const preHeat = (translations, format) => (
   reduce(translations, (acc, o, language) => {
-    acc[language] = reduce(o, (acc, v, k) => {
-      acc[k] = new IntlFormat(v, kebabCase(language), {...moneyFormat(language), ...format})
-      return acc
-    }, {})
+    acc[language] = preheatValues(o, format, language)
     return acc
   }, {})
 )
 
-const templateReducer = (values) => (acc, v, k) => {
-  acc[k] = v.format(values)
+const preheatValues = (obj, format, language) => {
+  return reduce(obj, (acc, value, key) => {
+    if (typeof value === 'object') {
+      acc[key] = preheatValues(value, format, language)
+    } else {
+      acc[key] = new IntlFormat(value, kebabCase(language), {...moneyFormat(language), ...format})
+    }
+    return acc
+  }, {})
+}
+
+const templateReducer = (values) => partial(templateIteratee, values)
+const templateIteratee = (values, acc, v, k) => {
+  if (typeof v === 'object' && !v.format) {
+    acc[k] = reduce(v, partial(templateIteratee, values), {})
+  } else {
+    acc[k] = v.format(values)
```

Comment:
Yeah, that's reasonable, should probably have a default guard. In practice, there are exactly two inputs that will make it into this function as `v`:

1. An instance of `IntlFormat` for the normal case, a string translation that's been pre-warmed to do Intl templating (e.g. money, date formatting etc)
2. My additional case of a sub-object of values, where each sub-value will either be said instance of `IntlFormat` from the prior pre-warming step, or another sub-object (although I doubt we'd ever need to go more than one level deep).